Formula/kube-aws.rb

```diff
@@ -14,44 +14,72 @@ class KubeAws < Formula
-      system "go", "generate", "./pkg/config"
-      system "go", "build", "-ldflags",
-             "-X github.com/coreos/coreos-kubernetes/multi-node/aws/pkg/cluster.VERSION=#{version}",
-             "-a", "-tags", "netgo", "-installsuffix", "netgo",
-             "-o", bin/"kube-aws", "./cmd/kube-aws"
+    if build.devel? || build.head?
+      cd kube_aws_dir do
+        system "go", "generate", "./config"
+        system "go", "generate", "./nodepool/config"
+        system "go", "build", "-ldflags",
+               "-X github.com/coreos/kube-aws/cluster.VERSION=#{version}",
+               "-a", "-tags", "netgo", "-installsuffix", "netgo",
+               "-o", bin/"kube-aws", "./"
+      end
+    else
+      cd kube_aws_dir do
+        system "go", "generate", "./config"
+        system "go", "build", "-ldflags",
+               "-X github.com/coreos/kube-aws/cluster.VERSION=#{version}",
+               "-a", "-tags", "netgo", "-installsuffix", "netgo",
+               "-o", bin/"kube-aws", "./cmd/kube-aws"
+      end
     end
   end
 
   test do
-    system "#{bin}/kube-aws"
-
-    cluster = { "clusterName" => "test-cluster", "externalDNSName" => "dns",
-                "keyName" => "key", "region" => "west",
-                "availabilityZone" => "zone", "kmsKeyArn" => "arn" }
-    system "#{bin}/kube-aws", "init", "--cluster-name", "test-cluster",
-           "--external-dns-name", "dns", "--region", "west",
-           "--availability-zone", "zone", "--key-name", "key",
-           "--kms-key-arn", "arn"
-    cluster_yaml = YAML.load_file("cluster.yaml")
-    assert_equal cluster, cluster_yaml
-
-    installed_version = shell_output("#{bin}/kube-aws version 2>&1")
-    assert_match "kube-aws version #{version}", installed_version
+    if build.devel? || build.head?
+      system "#{bin}/kube-aws"
+      cluster = { "clusterName" => "test-cluster", "externalDNSName" => "dns",
+                  "keyName" => "key", "region" => "west",
+                  "availabilityZone" => "zone", "kmsKeyArn" => "arn", "controller" => nil }
+      system "#{bin}/kube-aws", "init", "--cluster-name", "test-cluster",
+             "--external-dns-name", "dns", "--region", "west",
+             "--availability-zone", "zone", "--key-name", "key",
+             "--kms-key-arn", "arn"
+      cluster_yaml = YAML.load_file("cluster.yaml")
+      assert_equal cluster, cluster_yaml
+    else
+      system "#{bin}/kube-aws"
+      cluster = { "clusterName" => "test-cluster", "externalDNSName" => "dns",
+                  "keyName" => "key", "region" => "west",
+                  "availabilityZone" => "zone", "kmsKeyArn" => "arn" }
+      system "#{bin}/kube-aws", "init", "--cluster-name", "test-cluster",
+             "--external-dns-name", "dns", "--region", "west",
+             "--availability-zone", "zone", "--key-name", "key",
+             "--kms-key-arn", "arn"
+      cluster_yaml = YAML.load_file("cluster.yaml")
+      assert_equal cluster, cluster_yaml
```

Comment:
Similarly a bunch of the above can be put outside the `if/else`. Can you also share the parts of the hash where possible? Thanks!